core/field.js

```diff
@@ -51,15 +51,55 @@ goog.require('Blockly.utils.style');
  * @constructor
  */
 Blockly.Field = function(value, opt_validator, opt_config) {
+  /**
+   * A generic value possessed by the field.
+   * Should generally be non-null, only null when the field is created.
+   * @type {*}
+   * @protected
+   */
+  this.value_ = null;
+
+  /**
+   * Whether or not the field's value has been initialized.
+   * @type {!boolean}
+   * @private
+   */
+  this.valueInitialized_ = false;
+
+  /**
+   * The initial value to be set when the field is initialized.
+   * @type {*}
+   * @private
+   */
+  this.initialValue_ = value;
+
+  /**
+   * Validation function called when user edits an editable field.
```

Comment:
Nit: Does this need doc updates to match initialValue_?

Reply:
Yep!